src/test/java/edu/cornell/kfs/concur/service/impl/ConcurEventNotificationServiceImplTest.java

```diff
@@ -64,5 +67,39 @@ public void validateOverageRespone() {
     private String buildLongMessage() {
         return StringUtils.repeat("A", ConcurConstants.VALIDATION_RESULT_MESSAGE_MAX_LENGTH);
     }
+    
+    @Test
+    public void findConcurFailedEventQueueProcessingModeReadOnly() {
+        validateConcurFailedEventQueueProcessingController(ConcurFailedEventQueueProcessingMode.READONLY, 
+                findConcurFailedEventQueueProcessingControllerByString(ConcurConstants.CONCUR_FAILED_EVENT_QUEUE_READONLY));
+    }
+    
+    @Test
+    public void findConcurFailedEventQueueProcessingModeReadWrite() {
+        validateConcurFailedEventQueueProcessingController(ConcurFailedEventQueueProcessingMode.READWRITE, 
+                findConcurFailedEventQueueProcessingControllerByString(KFSConstants.ParameterValues.YES));
+    }
+    
+    @Test
+    public void findConcurFailedEventQueueProcessingModeOff() {
+        validateConcurFailedEventQueueProcessingController(ConcurFailedEventQueueProcessingMode.OFF, 
+                findConcurFailedEventQueueProcessingControllerByString(KFSConstants.ParameterValues.NO));
+    }
+    
+    @Test
+    public void findConcurFailedEventQueueProcessingModeDefault() {
+        validateConcurFailedEventQueueProcessingController(ConcurFailedEventQueueProcessingMode.OFF, 
+                findConcurFailedEventQueueProcessingControllerByString("foo"));
+    }
+    
+    private ConcurFailedEventQueueProcessingMode findConcurFailedEventQueueProcessingControllerByString(String processFailedEventQueue) {
```

Comment:
This helper method should be renamed to reflect the enum's "Controller"-to-"Mode" name change.